sdk/storage/azure-storage-blob/tests/perfstress_tests/upload.py

```diff
@@ -0,0 +1,28 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from ._test_base import _BlobTest
+
+from azure_devtools.perfstress_tests import RandomStream, get_random_bytes
+from azure_devtools.perfstress_tests import AsyncRandomStream
+
+
+class UploadTest(_BlobTest):
```

Comment:
I think your Upload and Download tests have the pattern reversed.  For Dowload, all instances can use the same blob in the service.  For upload, each instance should upload to its own blob.  In general, for read-only it's OK to share, but we should avoid concurrent writes to the same resource.

Reply:
I think I have the pattern correct (what you describe is what it should be doing) - as use of the base class `_BlobTest` indicates a unique blob to that instance.

Reply:
You are correct, somehow _I_ got this reversed when reading the code.